compressor/tests/test_base.py

```diff
@@ -332,6 +335,52 @@ def test_get_cachekey_leading_dot(self):
     def test_get_cachekey_invalid_mod(self):
         self.assertRaises(ImportError, lambda: get_cachekey("foo"))
 
+
+@override_settings(
+    COMPRESS_ENABLED=True,
+    COMPRESS_PRECOMPILERS=(),
+    COMPRESS_DEBUG_TOGGLE='nocompress',
+    DEBUG=True
+)
+class CompressorInDebugModeTestCase(SimpleTestCase):
+
+    def setUp(self):
+        self.css = '<link rel="stylesheet" href="/static/css/one.css" type="text/css" />'
+        self.tmpdir = mkdtemp()
+        copytree(settings.STATIC_ROOT, os.path.join(self.tmpdir, "static"))
+        self.old_compress_root = settings.COMPRESS_ROOT
+        settings.STATIC_ROOT = settings.COMPRESS_ROOT = os.path.join(self.tmpdir, "static")
+        settings.STATICFILES_DIRS = [self.old_compress_root]
+
+    def tearDown(self):
+        rmtree(self.tmpdir)
+        settings.STATIC_ROOT = settings.COMPRESS_ROOT = self.old_compress_root
+        delattr(settings, "STATICFILES_DIRS")
```

Comment:
it actually threw an attribute error when i tried that, so there's why the original PR did that :)

instead i had another look at how the context manager was used in the OfflineTestCaseMixin, finally understood it and tried that. seems to work well.